src/Riok.Mapperly/Descriptors/MappingBuilders/EnumToStringMappingBuilder.cs

```diff
@@ -16,8 +16,14 @@ public static class EnumToStringMappingBuilder
         if (ctx.Target.SpecialType != SpecialType.System_String || !ctx.Source.IsEnum())
             return null;
 
+        return BuildEnumToStringMapping(ctx);
+    }
+
+    private static EnumToStringMapping BuildEnumToStringMapping(MappingBuilderContext ctx)
+    {
+        var customNameMappings = ctx.Source.BuildCustomNameStrategyMappings(ctx.Configuration.Enum.NamingStrategy);
         // to string => use an optimized method of Enum.ToString which would use slow reflection
         // use Enum.ToString as fallback (for ex. for flags)
-        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source));
+        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source), customNameMappings);
```

Comment:
I'm thinking about whether it would make sense together with #1483 instead of figuring out which members is mapped how (to what value) in the mapping, doing it here and passing a `IReadOnlyList<EnumMemberMapping>` which consists of two `Expression` one for the source and one for the target, wdyt?

Reply:
Yes, considering that the current implementation involves multiple collections of explicit mappings that will need to be handled later, it would likely be better to create a single collection of mappings in one place.
It might be best to wait until #1483 is merged before making these changes in this branch.